tempesta_fw/sock.c

```diff
@@ -340,6 +340,23 @@ ss_sock_active(struct sock *sk)
  *  	Server and client connections handling
  * ------------------------------------------------------------------------
  */
+/**
+ * The simplified version of sk_forced_mem_schedule().
+ * We use the forced version of the socket accounting function to send the
+ * data ASAP - we can't wait for memory as process context does.
```

Comment:
> we can't wait for memory

As far as I understand, the only difference from `sk_forced_mem_schedule()` is that latter also calls `mem_cgroup_charge_skmem()`. Looks like `mem_cgroup_charge_skmem()` is aware of being called from softirq context:
```c
        /* Don't block in the packet receive path */
        if (in_softirq())
                gfp_mask = GFP_NOWAIT;
```
So it looks like waiting is not an issue?

Reply:
Yeah. Firstly, the function is quite small and I though that it makes sense just to copy it instead of to patch the kernel with the function exporting. Also the function doesn't make the unnecessary `mem_cgroup_charge_skmem()` call - not a big deal to call it, but all in all it makes sense to make the short simplified copy of the function.

Reply:
Oh, right. I missed that `sk_forced_mem_schedule()` is not exported.